source/clear-linux/tutorials/kubernetes.rst

```diff
@@ -99,111 +106,104 @@ deployment and your security needs.
+    locate the master's host:
 
-   a) Run Kubernetes with Docker + runc:
+    .. code-block:: bash
 
-      #. Enable docker.service:
+       echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
 
-         .. code-block:: bash
+#.  Exit root:
 
-            sudo systemctl enable docker.service
+    .. code-block:: bash
 
-      #. Create (or edit if it exists) the file 
-         :file:`/etc/systemd/system/docker.service.d/51-runtime.conf` and include the following lines:
+       exit
 
-         .. code-block:: bash
+Configure and run Kubernetes
+****************************
 
-            [Service]
-            Environment="DOCKER_DEFAULT_RUNTIME=--default-runtime runc"
+This section describes how to configure and run Kubernetes with:
 
-      #. Create (or edit if it exists) the file :file:`/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` and include the following lines:
+* Docker and runc
+* CRI-O and kata-runtime
 
-         .. code-block:: bash
+Configure and run Docker + runc
+===============================
 
-            [Service]
-            Environment="KUBELET_EXTRA_ARGS="
+#.  Enable the Docker service:
 
-      #. Restart services:
+    .. code-block:: bash
 
-         .. code-block:: bash
+       sudo systemctl enable docker.service
 
-            sudo systemctl daemon-reload
-            sudo systemctl restart docker
-            sudo systemctl restart kubelet
+#.  Create (or edit if it exists) the file 
+    :file:`/etc/systemd/system/docker.service.d/51-runtime.conf` and include the following lines:
 
-   or:
+    .. code-block:: bash
 
-   b) Run Kubernetes with CRI-O + your desired runtime. You can use multiple
-      runtimes with CRI-O, including *runc* and *kata-runtime*. To use
-      CRI-O + *kata-runtime*:
+       [Service]
+       Environment="DOCKER_DEFAULT_RUNTIME=--default-runtime runc"
 
-      #. Enable crio.service:
+#.  Create (or edit if it exists) the file :file:`/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` and include the following lines:
 
-         .. code-block:: bash
+    .. code-block:: bash
 
-            sudo systemctl enable crio.service
+       [Service]
+       Environment="KUBELET_EXTRA_ARGS="
 
-      #. Restart services:
+#.  Enter the commands:
 
-         .. code-block:: bash
+    .. code-block:: bash
 
-            sudo systemctl restart crio
-            sudo systemctl restart kubelet
+       sudo systemctl daemon-reload
+       sudo systemctl restart docker
+       sudo systemctl restart kubelet
 
-Run Kubernetes for the first time
-*********************************
+#.  Initialize the master control plane with the command:
 
-#. Prepare your system to run Kubernetes for the first time with the
-   following commands, either:
+    .. code-block:: bash
 
-   a) If you are running Kubernetes with Docker + runc:
+       sudo kubeadm init --pod-network-cidr 10.244.0.0/16 --ignore-preflight-errors=SystemVerification
```

Comment:
```--pod-network-cidr 10.244.0.0/16``` is only needed if Flannel CNI is used. We can leave this out from the generic instructions.

Reply:
I tested this, and yes, we don't need that extra command. Bun and I tested it.  @mythi Thank you.

Reply:
thanks. @MCamp859 please drop that parameter from all commands it's used.

Reply:
@mythi, @joselamego and I just tested whether or not one should use the flag `--pod-network-cidr 10.244.0.0/16` with *CRI-O + kata-runtime*, and when I tested it without this flag it failed. That leads me to believe this flag is necessary when using CRI-O + kata-runtime. 

@joselamego Please advise if different than above.

Reply:
@mvincerx: CNI is the default network plugin in Clear Linux, so it should be part of the CRI-O instructions

Reply:
@mvincerx what is is that failed?

Reply:
@mythi, @mvincerx, Clarifying: As @mythi said, the flag `--pod-network-cidr` is only needed in the CRI-O + kata instructions, not in the Docker + runc section. We tested the former.

Reply:
Here's what I said: 
> ```--pod-network-cidr 10.244.0.0/16``` is only needed if Flannel CNI is used.

Reply:
Correct, and CNI is default for CRI-O in Clear Linux

Reply:
CRI-O does not have CNI plugin(s) configured by default.

Reply:
@joselamego it's safe to leave out but I think we need to add few separate notes about CNI setup to remind readers where special config/parameters are needed.

Reply:
@mythi, please include comments here, or send to @MCamp859

Reply:
I would suggest from my experience in following this tutorial that you could point to Kubernetes' documentation around pod networks. `10.244.0.0/16` should be explained as it is specific to `flannel`. (just my 2 cents)

Reply:
To improve the tutorial wrt networking, my proposal is we add item 3 under "Run Kubernetes for the first time" that guides user to pick and install a CNI plugin using https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#pod-network . It should also add a note that user carefully checks the plugin specific extra steps that might be required (e.g., ```--pod-network-cidr 10.244.0.0/16``` for flannel). Also weave plugin installs itself in ```/opt/cni/bin``` which requires that CRI-O's or kubelet's (depending on whether CRI-O or Docker is used) CNI plugin dir parameter is changed accordingly.

Reply:
I support @mythi's proposal. Let's provide the `kubeadm init` command without flags and add the CNI plugins note and link.